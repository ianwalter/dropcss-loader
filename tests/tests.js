const { join } = require('path')
const { test } = require('@ianwalter/bff')
const webpack = require('webpack')
const MemoryFileSystem = require('memory-fs')

test('unused CSS rules are removed', ({ expect }) => {
  return new Promise (resolve => {
    const compiler = webpack({
      entry: join(__dirname, './fixtures/example.css'),
      module: {
        rules: [
          {
            test: /\.css$/,
            use: [
              {
                loader: '@ianwalter/callback-loader',
                options: {
                  callback (content) {
                    expect(content).toMatchSnapshot()
                    resolve()
                  }
                }
              },
              {
                loader: join(__dirname, '../'),
                options: { html: join(__dirname, 'fixtures/example.html') }
              }
            ]
          }
        ]
      }
    })
    compiler.outputFileSystem = new MemoryFileSystem()
    compiler.run()
  })
})
