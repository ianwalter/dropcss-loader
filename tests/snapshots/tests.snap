// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unused CSS rules are removed 1`] = `"h2.example{font-size: 2rem;}h5.example{font-size: 1.25rem;}"`;
